cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -61,6 +89,32 @@ size_t getContingencyMatrixWorkspaceSize(int nSamples,
     nSamples, groundTruth, stream, minLabel, maxLabel);
 }
 
+/**
+ * @brief Calculate workspace size for running contingency matrix calculations
+ * @tparam DataT label type
+ * @tparam IdxType Index type of matrix extent.
+ * @param handle: the raft handle.
+ * @param groundTruth: device 1-d array for ground truth (num of rows)
+ * @param minLabel: Optional, min value in input array
+ * @param maxLabel: Optional, max value in input array
+ */
+template <typename DataT, typename IdxType>
+size_t getContingencyMatrixWorkspaceSize(const raft::handle_t& handle,
```

Comment:
We should not expose workspaces to the public API. The original functions were written before we had RMM (and could just use a pool allocator).

Reply:
This one still needs to be removed from the public API. Rather than having `contingency_matrix` accept a workspace, please have it compute and allocate the workspace internally. Eventually we will have an optional rmm device memory resources attached to the raft handle that can be used to override the allocation strategy.